hts_internal.h

```diff
@@ -120,6 +120,22 @@ int bgzf_idx_push(BGZF *fp, hts_idx_t *hidx, int tid, hts_pos_t beg, hts_pos_t e
  */
 void bgzf_idx_amend_last(BGZF *fp, hts_idx_t *hidx, uint64_t offset);
 
+static inline int find_file_extension(const char *fn, char ext_out[5])
+{
+    const char *delim = fn ? strstr(fn, HTS_IDX_DELIM) : NULL, *ext;
+    if (!fn) return -1;
+    if (!delim) delim = fn + strlen(fn);
+    for (ext = delim; ext > fn && *ext != '.' && *ext != '/'; --ext) {}
+    if (*ext == '.' && delim - ext == 3 && ext[1] == 'g' && ext[2] == 'z') {
+        // permit .sam.gz as a valid file extension
```

Comment:
There's no specific check for `.bgz` here, which has issues for `.vcf.bgz` detection.

If we add it as a specific case, `ext_out` would need to become `ext_out[static 8]` (or `.bgz` could be converted to `.gz` when filling `ext_out`).